nixos/modules/hardware/opengl.nix

```diff
@@ -129,32 +156,35 @@ in
   };
 
   config = mkIf cfg.enable {
-    assertions = [
-      { assertion = cfg.driSupport32Bit -> pkgs.stdenv.isx86_64;
+    assertions = mkIf cfg.opengl.enable [
+      { assertion = cfg.opengl.driSupport32Bit -> pkgs.stdenv.isx86_64;
         message = "Option driSupport32Bit only makes sense on a 64-bit system.";
       }
-      { assertion = cfg.driSupport32Bit -> (config.boot.kernelPackages.kernel.features.ia32Emulation or false);
+      { assertion = cfg.opengl.driSupport32Bit -> (config.boot.kernelPackages.kernel.features.ia32Emulation or false);
         message = "Option driSupport32Bit requires a kernel that supports 32bit emulation";
       }
     ];
 
     systemd.tmpfiles.rules = [
-      "L+ /run/opengl-driver - - - - ${package}"
+      "L+ ${impurePath} - - - - ${package}"
       (
         if pkgs.stdenv.isi686 then
-          "L+ /run/opengl-driver-32 - - - - opengl-driver"
-        else if cfg.driSupport32Bit then
-          "L+ /run/opengl-driver-32 - - - - ${package32}"
+          "L+ ${impurePath32} - - - - ${impurePath}"
```

Comment:
The target used to be a relative path (`"opengl-driver"`), now it's absolute (`"/run/opengl-driver"`). Shouldn't matter